go-controller/pkg/ovn/gateway_init.go

```diff
@@ -68,10 +68,34 @@ func (oc *Controller) gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet,
 		},
 	}
 
+	var oldExtIPs []net.IP
+	// If l3gatewayAnnotation.IPAddresses changed, we need to update the perPodSNATs,
+	// so let's save the old value before we update the router for later use
+	oldlogicalGRRes := []nbdb.LogicalRouter{}
+	ctx, cancel := context.WithTimeout(context.Background(), types.OVSDBTimeout)
+	defer cancel()
+	if err := oc.nbClient.WhereCache(func(lr *nbdb.LogicalRouter) bool {
+		return lr.Name == gatewayRouter
+	}).List(ctx, &oldlogicalGRRes); err != nil {
+		return fmt.Errorf("failed in retrieving %s, error: %v", gatewayRouter, err)
+	}
+	// no need to do anything if GR doesn't exist yet
+	if len(oldlogicalGRRes) > 0 {
+		oldExternalIPs := strings.Split(oldlogicalGRRes[0].ExternalIDs["physical_ips"], ",")
+		oldExtIPs = make([]net.IP, len(oldExternalIPs))
+		for i, oldExternalIP := range oldExternalIPs {
+			cidr := oldExternalIP + GetIPFullMask(oldExternalIP)
+			ip, _, err := net.ParseCIDR(cidr)
+			if err != nil {
+				return fmt.Errorf("failed to update GW SNAT rule for pods on router %s error: %v", gatewayRouter, err)
```

Comment:
this is really an IP parsing failure. Shouldn't you print the ip the mask which failed to be parsed?

Reply:
++ you are right.